### ISSUE TITLE
Slider field fails to take ephemeral focus

### ISSUE THREAD

**Issue by rachel-fenichel** [rachel-fenichel/blockly-samples#2514]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Component

`field-slider`

### Description

Error in console: attempted to take ephemeral focus when it's already held.

New issue with v12.

### Reproduction steps

Link against Blockly v12.0.0-beta.5 and open the plugin test page.

Click a slider field to open the editor.

### Stack trace

```shell

```

### Screenshots

![Image](https://github.com/user-attachments/assets/8e5e0751-65fa-4f75-86c6-de39f8f2a3e3)

**Comment by BenHenning** [comment]
This is caused by `FieldSlider` doing an unexpected thing: it creates its own drop-down div after initializing its parent's (`FieldInput`'s) inline div (which uses widget div). Blockly currently supports only one widget div or drop-down div at a time, but per type. It didn't occur to me that both could be opened simultaneously, so them both trying to take and manage ephemeral focus just doesn't work.

There's probably some thought that could go into how to properly synchronize multiple things wanting to manage ephemeral focus at the same time, but it's not straightforward since it isn't clear who gets the ability to restore.

A reasonable fix here would be to expand `FieldInput` and both widget/drop-down divs to support disabling their automatic ephemeral focus management so that there isn't interference. This provides more flexibility when using these, anyway, which seems like a net win.

**Comment by BenHenning** [comment]
I noticed a separate issue as well: changing the field value with the inline-editor doesn't automatically update the slider. This actually contrasts from results when testing #2515 in that the angle slider _does_ automatically update.

**Comment by BenHenning** [comment]
> I noticed a separate issue as well: changing the field value with the inline-editor doesn't automatically update the slider. This actually contrasts from results when testing [#2515](https://github.com/google/blockly-samples/issues/2515) in that the angle slider _does_ automatically update.

Ah! This is an existing problem, actually. Filed #2520 to track.

**Comment by BenHenning** [comment]
Fixed in https://github.com/google/blockly-samples/pull/2521.
